### ISSUE TITLE
LCD Character Display Code Block Assumes I2C 0x27 Address

### ISSUE THREAD

**Issue by sk8board** [BIPES/BIPES#227]
The LCD character display code block assumes the I2C address is 0x27. If your display has a different address than 0x27, then the LCD block will not function for your display.

The LCD code block needs to allow the user to change the I2C address as needed. Therefore, the LCD block would need to add another input for I2C address.

With that said, there should be an additional code block that can scan for I2C addresses so the user can find the address of their LCD display.

![image](https://github.com/BIPES/BIPES/assets/96895142/5041a029-0e76-4a66-915a-36b6f370ec81)

![image](https://github.com/BIPES/BIPES/assets/96895142/f895967f-2adf-4ada-ab1f-b2ba290f5e54)

**Comment by jetannenbaum** [comment]
@sk8board: Do you have a specific board in mind that uses a different address?  Also, adding an I2C scan() method, while trivial, gets us right back into the discussion of the I2C init(...) method.  

From what I can see, it looks like one of the libraries (lcd_api) isn't loaded.  Did you load both the lcd_api & pico_i2c_lcd libraries?

**Comment by sk8board** [comment]
@jetannenbaum Yes, I loaded both libraries. The problem is my display has an address of 0x3F but the block assumes 0x27.

**Comment by jetannenbaum** [comment]
@sk8board:  Seems like the code should scan for the address, rather than assume that the address is 0x27.  That would be a trivial change inside the I2C Character LCD code (char_lcd_init block).  Want to take a swing at modifying the code?

**Comment by sk8board** [comment]
@jetannenbaum I am attempting to fix this by adding another input field which would allow the user to choose a different address. 

Unfortunately, I discovered Blockly does not support hexadecimal numbers with their number fields. Therefore, I tried to use a text field, but a text field looks out of place as well as the text variable gets imported into MicroPython with '3F ' apostrophe marks.

![image](https://github.com/BIPES/BIPES/assets/96895142/08d37902-a9ce-479d-8876-4bb6b0676323)

![image](https://github.com/BIPES/BIPES/assets/96895142/4d47f982-04f8-4891-9ef9-cffd5fb42894)

I like your idea. I will give it a try.
